ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/messenger/messagelistpage/MessageListContainerFragmentTest.java

```diff
@@ -0,0 +1,403 @@
+
+  @Mock
+  private Context context;
+
+  @Mock
+  private MessageListContainerContract.Data contractData;
+
+  @Mock
+  private LayoutInflater inflater;
+
+  @Mock
+  private ViewGroup viewGroup;
+
+  @Mock
+  private View view;
+
+  @Mock
+  private FragmentActivity fragmentActivity;
+
+  @Mock
+  private Intent intent;
+
+  @Mock
+  private Resources resources;
+
+  @Rule
+  private ExpectedException thrown = ExpectedException.none();
+
+  @Before
+  public void setUp() {
+    messageListContainerFragment = PowerMockito.spy(new MessageListContainerFragment());
+    PowerMockito.mockStatic(HelpCenterPref.class);
+    PowerMockito.mockStatic(MessengerPref.class);
+    PowerMockito.mockStatic(Kayako.class);
+    PowerMockito.mockStatic(MessageListContainerFactory.class);
+    PowerMockito.mockStatic(FileAttachmentUtil.class);
+    PowerMockito.mockStatic(Toast.class);
+    PowerMockito.mockStatic(Gravity.class);
+    Mockito.when(messageListContainerFragment.getActivity()).thenReturn(fragmentActivity);
+    Mockito.when(messageListContainerFragment.getContext()).thenReturn(context);
+    Mockito.when(fragmentActivity.getIntent()).thenReturn(intent);
+    Mockito.when(intent.getExtras()).thenReturn(bundle);
+  }
+
+  @Test
+  public void callRetainInstanceOnCreate() {
+    // Arrange
+    PowerMockito.when(MessageListContainerFactory.getData())
+        .thenReturn(contractData);
+
+    // Act
+    messageListContainerFragment.onCreate(bundle);
+
+    // Assert
+    Mockito.verify(messageListContainerFragment).onCreate(bundle);
+    Mockito.verify(messageListContainerFragment).setRetainInstance(true);
+    PowerMockito.verifyStatic(MessageListContainerFactory.class, Mockito.times(1));
+    MessageListContainerFactory.getPresenter(messageListContainerFragment);
+  }
+
+  @Test
+  public void callInflateOnCreateView() {
+    // Arrange
+
+    // Act
+    messageListContainerFragment.onCreateView(inflater, viewGroup, null);
+
+    // Assert
+    Mockito.verify(inflater).inflate(R.layout.ko__fragment_message_list_container, viewGroup, SAVED_INSTANCE_STATE);
+  }
+
+  @Test
+  public void fillFieldsOnViewCreated() {
+    // Arrange
+
+    // Act
+    messageListContainerFragment.onViewCreated(view, bundle);
+
+    // Assert
+    Mockito.verify(messageListContainerFragment).onViewCreated(view, bundle);
+    Mockito.verify(messageListContainerFragment, Mockito.times(3)).getChildFragmentManager();
+  }
+
+  @Test
+  public void callOnActivityCreatedWithNonNullBundle() {
+    // Arrange
+    MessageListContainerFragment fragmentMock = PowerMockito.mock(MessageListContainerFragment.class);
+    bundle.putLong(KayakoSelectConversationActivity.ARG_CONVERSATION_ID, 1L);
+    Mockito.when(fragmentMock.getActivity()).thenReturn(fragmentActivity);
+
+    // Act
+    fragmentMock.onActivityCreated(bundle);
+
+    // Assert
+    Mockito.verify(fragmentMock).onActivityCreated(bundle);
+  }
+
+  @Test
+  public void testOnActivityResultWithAddAttachment() {
```

Comment:
⚠️  The `test` prefix is specific to JUnit 3, while `@Test` is specific to JUnit 4 and deprecates the former. It could be a good approach to remove this prefix and avoid redundancy.   

**[More details](https://confluence.devfactory.com/x/TAJrEw)**
          
## The possible solution in this case

```JAVA
@Test 
public void onActivityResultWithAddAttachment() {

```
 

<a id="b74ec9c0-4f1d-11e8-b712-a55fdd633c3d" href="none" codereview-uuid-reserved-tag />